packages/ckeditor5-table/src/tablecolumnresize/commands/tableresizecolumncommand.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license Copyright (c) 2003-2024, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module table/tablecolumnresize/commands/tableresizecolumncommand
+ */
+
+import { Command, type Editor } from 'ckeditor5/src/core.js';
+import type { DowncastInsertEvent } from 'ckeditor5/src/engine.js';
+import type { PossibleResizeColumnRange } from '../tablecolumnresizeutils.js';
+import { _getSmallestColumnResizer } from '../utils/getsmallestcolumnresizer.js';
+
+/**
+ * The resize table column command.
+ *
+ * The command is registered by {@link module:table/tablecolumnresize/tablecolumnresizeui~TableColumnResizeUI}
+ * as the `'resizeTableColumn'` editor command.
+ *
+ * To resize currently selected column, execute the command:
+ *
+ * ```ts
+ * editor.execute( 'resizeTableColumn', { newColumnWidth: 250 } );
+ * ```
+ */
+export default class TableResizeColumnCommand extends Command {
+	/**
+	 * The command value: Current size of column (in pixels) and possible resize range of such column..
+	 *
+	 * @readonly
+	 * @observable
+	 */
+	declare public possibleRange: PossibleResizeColumnRange | null;
```

Comment:
It can be, but it simplifies a lot of the code due to the fact that the currently selected column width is calculated alongside with range, and knowing the fact that the calculation of column widths is quite expensive it's good enough to stay as it is.